### PR TITLE
docs: add general documentation about vpn protocols

### PR DESCRIPTION
**Pull request by lemoer**

### CODE REVIEW

File: docs/features/vpn.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
+VPN
+===
+
+Gluon supports different options to establish vpn tunnels,
+which connect mesh clouds and provide internet access.
```
Comment (rotanid) [inline docs/features/vpn.rst]: not necessarily internet...

File: docs/features/vpn.rst [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Currently the available vpn protocols options are:
```
Comment (rotanid) [inline docs/features/vpn.rst]: hm, is it correct to have both "protocols" and "options" in plurar?

File: docs/features/vpn.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+- fastd
+- L2TP (tunneldigger)
+
+Fastd is a lightweight vpn daemon in userspace, which is
+especially designed for embedded hardware. It supports
+encryption and authentication.
+
+L2TP is implemented inside the linux kernel and has
```
Comment (rotanid) [inline docs/features/vpn.rst]: has therefore -> therefore has

File: docs/features/vpn.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+therefore performance advantages over fastd. The
+disadvantage of L2TP is, that it does not support any
+encryption. So everything is sent in plain.
```
Comment (rotanid) [inline docs/features/vpn.rst]: last sentence redundant?
also: 2nd disadvantage: currently tunneldigger doesn't support IPv6

File: docs/features/vpn.rst [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+Optional Encryption (fastd only):
+---------------------------------
+
+When using fastd, the firmware can allow the user to
+decide by itself, whether he want's to use encryption
```
Comment (rotanid) [inline docs/features/vpn.rst]: *himself
*wants

File: docs/features/vpn.rst [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+or not. If the firmware builder doesn't like this, he
+is also able to hide (or even forbid) the encryptionless
```
Comment (rotanid) [inline docs/features/vpn.rst]: hide ... from (not to)
encryptionless? sounds strange...

File: docs/features/vpn.rst [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+option to the user.
+
+If you want to allow users to decide by themselves:
+
+- Be sure, the package ``gluon-web-mesh-vpn-fastd`` is enabled in ``site.mk``
+- Set the option ``mesh_vpn.fastd.configurable = true`` in ``site.conf``
+- On the server side, be sure that ``null`` cipher is allowed and preferred over ``salsa2012+umac``. You can ensure this by inserting the ``method "null";`` entry before the ``method "salsa2012+umac";`` in your fastd.conf.
```
Comment (rotanid) [inline docs/features/vpn.rst]: "be sure that" -> "be sure the"

File: docs/features/vpn.rst [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+Users now should have the choice in expert mode to decide
+by themselves, which looks like this:
+
+.. image:: fastd_mode.gif
+
+If you want to ensure, that the correct chipher is chosen,
```
Comment (rotanid) [inline docs/features/vpn.rst]: maybe better: "If you want to check if the correct ..."

File: docs/features/vpn.rst [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+you can use the following command on a router. You maybe
```
Comment (rotanid) [inline docs/features/vpn.rst]: maybe "Gluon node" instead of "router" ?

**Comment by lemoer** [comment]
Feel free to change here: https://md.darmstadt.ccc.de/gluon-pr_doc_vpn

File: docs/features/vpn.rst (hunk 1 above)
Comment (mweinelt) [reply docs/features/vpn.rst]: > internetwork access

the problem with internet is … it's a shorthand for internetwork … but by now it's meaning has been superseded by the one global network.

File: docs/features/vpn.rst (hunk 8 above)
Comment (rotanid) [reply docs/features/vpn.rst]: oh, and "chipher" -> "cipher" of course

**Comment by rotanid** [comment]
i updated the linked doc

**Comment by mweinelt** [comment]
lgtm

**Pull request merged**